pwlwp/matcher.py

```diff
@@ -8,13 +8,35 @@
 
 class SentenceMatch():
 
-	def __init__(self, sentence_embedding, data_embedding, threshold: float, is_causal: bool, is_belief: bool) -> None:
+	choice_str = ""
+
+	def __init__(self, choice_str, sentence_embedding, data_embedding, threshold: float, is_causal: bool, is_belief: bool) -> None:
 		similarity = self.similarity(sentence_embedding, data_embedding)
 		hit = threshold < similarity
 		self.all   =hit and True
 		self.causal=hit and is_causal
 		self.belief=hit and is_belief
 		self.both  =hit and is_causal and is_belief
+		self.choice_str = choice_str
+
+	def similarity(self, left_embedding, right_embedding) -> float:
+		result = numpy.dot(left_embedding, right_embedding) / numpy.linalg.norm(left_embedding) / numpy.linalg.norm(right_embedding)
+		return result
+
+class SentenceMatchFilter():
+
+	choice_str = ""
```

Comment:
Is this a strategy to guard against uninitialized variables or to make the types more obvious or something?  There are similar sentence_matches and choice_matches later.  I'm not familiar with the practice.

Reply:
My code is a bit messy and I probably have to clean it, but I created 2 SentenceMatch filters depending on wether we want to first filter by the introduction or not. I could've implemented it as a Boolean variable but that would've led to more complex code. Changing to a single class is pretty easy if you want me to change it though.

Reply:
It's just that `choice_str = ""` and a couple of similar lines that have me curious.  It doesn't seem like they are strictly necessary, so they probably serve an unfamiliar purpose.  I wouldn't change anything.  It's still research and experimental.  Maybe you can fill me in at the meeting.

Reply:
Oh I thought you're referring to the class as a whole. Well, "choice_str = """ is probably useless actually whops.